lib/ecto/migration.ex

```diff
@@ -613,7 +613,20 @@ defmodule Ecto.Migration do
     add(column, :naive_datetime, opts)
   end
 
-  def add(column, type, opts) when is_atom(column) do
+  def add(column, type, opts) when is_atom(column) and type in [:decimal, :numeric] do
```

Comment:
Right, I thought `precision` and `scale` only make sense for numerical type. Let me fix it real quick.